src/test/java/edu/oakland/test/database/DatabasePersistentStorageTest.java

```diff
@@ -0,0 +1,326 @@
+public class DatabasePersistentStorageTest {
+
+  String[] randomArray = {"A", "B"};
+
+  LocationDataPoint ldp = new LocationDataPoint(0, 0, LocalDateTime.of(
+        (int) (Math.random() * 50 + 1970),
+        (int) (Math.random() * 12 + 1),
+        (int) (Math.random() * 28 + 1),
+        (int) (Math.random() * 24),
+        (int) (Math.random() * 60)
+    ));
+
+
+  float course = generateRandomCourse();
+  TrackData trackDataPoint = new TrackData(
+        generateRandomLocationDataPointsArray(5),
+        course,
+        generateRandomSpeed()
+    );
+
+
+  ArrayList<LocationDataPoint> locDataPoint = new ArrayList<LocationDataPoint>();
+  ArrayList<TrackData> trackData = new ArrayList<TrackData>();
+
+  
+
+  @Test
+  @DisplayName("RFID data in is RFID data out")
+  void rfidDataInIsRfidDataOut() {
+    //int random = (int) (Math.random() * 60);
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+    DatabasePersistentStorageClass dpsc = new DatabasePersistentStorageClass(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+    assertEquals(0, dpsc.locateRfidData());
+  }
+
+  @Test
+  @DisplayName("offset matches data point")
+  void offsetMatchDataPoint() {
+    
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+    DatabasePersistentStorageClass dpsc = new DatabasePersistentStorageClass(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+    dpsc.storeLocationDataPoint(ldp);
+    assertEquals(ldp, dpsc.getLocationDataPoint(0));
+  }
+
+  @Test
+  @DisplayName("Location Data Point In Is Data Point Out")
+  void locationDataPointInIsDataPointOut() {
+
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+    DatabasePersistentStorageClass dpsc = new DatabasePersistentStorageClass(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+    int i = 0;
+    dpsc.storeLocationDataPoint(ldp);
+    assertEquals(ldp, dpsc.getLocationDataPoint(i));
+    
+  }
+
+  @Test
+  @DisplayName("Offset matches track data")
+  void offsetMatchTrackData() {
+    
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+    DatabasePersistentStorageClass dpsc = new DatabasePersistentStorageClass(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+
+
+    dpsc.storeTrackData(trackDataPoint);
+    assertEquals(trackDataPoint, dpsc.getTrackData(0));
+    
+  }
+
+
+  @Test
+  @DisplayName("Track Data In Is Track Data Out")
+  void trackDataInIsTrackDataOut() {
+    
+    locDataPoint.add(ldp);
+    trackData.add(trackDataPoint);
+    DatabasePersistentStorageClass dpsc = new DatabasePersistentStorageClass(0, 
+        locDataPoint, 
+        trackData, 
+        randomArray
+    );
+
+    dpsc.storeTrackData(trackDataPoint);
+    assertEquals(trackDataPoint, dpsc.getTrackData(0));
+
+
+  }
+
+  @Test
+  @DisplayName("read Mode In Is Mode Out")
+  void readModeInIsModeOut() {
```

Comment:
This test is redundant to the previous, so we should be able to just call it `modeInIsModeOut()` and delete the next test.
```suggestion
  void modeInIsModeOut() {
```